api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirObservationDaoImpl.java

```diff
@@ -40,6 +55,36 @@
 	@Autowired
 	private ObservationCategoryMap categoryMap;
 	
+	@Autowired
+	@Getter(AccessLevel.PUBLIC)
+	@Setter(AccessLevel.PUBLIC)
+	@Qualifier("sessionFactory")
+	private SessionFactory sessionFactory;
+	
+	@Override
+	@Authorized(PrivilegeConstants.GET_OBS)
+	public List<String> getSearchResultUuids(@Nonnull SearchParameterMap theParams) {
+		if (!theParams.getParameters(FhirConstants.LASTN_SEARCH_HANDLER).isEmpty()) {
+			
+			Criteria criteria = sessionFactory.getCurrentSession().createCriteria(typeToken.getRawType());
+			
+			setupSearchParams(criteria, theParams);
+			
+			criteria.setProjection(Projections.projectionList().add(Projections.property("uuid"))
+			        .add(Projections.property("concept")).add(Projections.property("obsDatetime")));
```

Comment:
Might be nice to import these as static imports, so this can just be written:

```java
criteria.setProjection(projectionList().add(property("uuid")).add(property("concept")).add(property("obsDateTime")));
```